src/ts/layers/circle.ts

```diff
@@ -38,12 +71,12 @@ module powerbi.extensibility.visual {
             const map = this.parent.getMap();
             const limits = this.source.getLimits();
             if (settings.circle.show) {
-                const sizes = Circle.getSizes(limits.size, map, settings, roleMap.size);
+                this.sizes = Circle.getSizes(limits.size, map, settings, roleMap.size);
 
                 let isGradient = mapboxUtils.shouldUseGradient(roleMap.color, limits.color);
                 let colors = Circle.getColors(limits.color, isGradient, settings, this.palette, roleMap.color);
 
-                map.setPaintProperty(Circle.ID, 'circle-radius', sizes);
+                map.setPaintProperty(Circle.ID, 'circle-radius', this.sizes);
```

Comment:
Why don't you set the circle-radius of the Circle.HighlightID layer here? That way we would not need to store the sizes array as a member and keep that in memory all the time.